### ISSUE TITLE
Is there some special options in allennlp to control training progress ?

### ISSUE THREAD

**Issue by robbine** [allenai/allennlp#2586]
Allennlp is a great NLP platform. Recently I am working on a classification problem. The weird thing that training accuracy is growing very fast, while validating accuracy increases very slowly.
Despite of that fact that distributions on training and validation varies a lot. I do shuffle the whole training data and instances in each batch. Here is the training configuration:
    "trainer": {
        "num_serialized_models_to_keep": 5,
        "cuda_device": [0,1],
        "grad_clipping": 1.0,
        "grad_norm": 1.0,
        "num_epochs": 30,
        "optimizer": {
            "type": "bert_adam",
            "lr": 0.00001
        },
        "patience": 3,
        "validation_metric": "+acc"
    }
My question is: Is there some special options in allennlp to control training progress ?

**Comment by matt-gardner** [comment]
If training accuracy goes up way more than validation accuracy does, that means you are overfitting your training set.  Some amount of this is expected in any machine learning problem, but if it is too severe, there's a problem with your model.  Some common solutions to this are to add some regularization to your model, add dropout (or more dropout), or decrease the number of parameters in your model.

I'm not sure what you mean by "control the training progress."  If you could be more specific, we may be able to help you better.

**Comment by robbine** [comment]
Thanks @matt-gardner !
I believe your solution s are what I am looking for.
For example by adding more dropout in order to reduce model complexity.
I am wondering how to add some regularization to my model, Does allennlp provide a way to do this ?

**Comment by matt-gardner** [comment]
Most models that we have implemented take a regularizer, which you can specify like this: https://github.com/allenai/allennlp/blob/9437b614dbc518b3b532bbaa201446dc454fc6e2/training_config/srl_elmo_5.5B.jsonnet#L43-L51

**Comment by robbine** [comment]
Thanks a lot ! I'll have a try.
